### PR TITLE
python3Packages.userpath: init at 1.4.1

### PR DESCRIPTION
**Pull request by yshym**
###### Motivation for this change

[userpath](https://github.com/ofek/userpath) is a cross-platform tool for adding locations to the user PATH. It is required to package a very useful python tool [pipx](https://github.com/pipxproject/pipx), which allows to install and run python applications in isolated environments.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/userpath/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  checkInputs = [ tox pytest coverage ];
+
+  # We should skip tox dependencies installation to run tests but
+  # tox doesn't have such an option yet (https://github.com/tox-dev/tox/issues/410)
+  doCheck = false;
+
+  checkPhase = ''
+    tox
+  '';
```
Comment (jonringer) [inline pkgs/development/python-modules/userpath/default.nix:30]: we don't need to run tox, nixpkgs already gives us per-interpreter tooling.

however, I looked at the test suite and couldn't make heads or tails of it.


Anyway:

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

Comment (yevhenshymotiuk) [reply pkgs/development/python-modules/userpath/default.nix]: Done.

File: pkgs/development/python-modules/userpath/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  version = "1.4.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256="0mfjmvx286z1dmnrc7bm65x8gj8qrmkcyagl0vf5ywfq0bm48591";
+  };
+
+  propagatedBuildInputs = [ click distro ];
+
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/userpath/default.nix:19]: it's still odd to see tests disabled, please give a comment as to why:

```
  # test suite is difficult to emulate in sandbox due to shell manipulation
```

Comment (yevhenshymotiuk) [reply pkgs/development/python-modules/userpath/default.nix:20]: Added.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 96114 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary><br>- python27Packages.userpath<br>- python37Packages.userpath<br>- python38Packages.userpath</details>

**Comment by yevhenshymotiuk** [comment]
Good, thx for the review

**Pull request merged**